parsons/databases/database/database.py

```diff
@@ -0,0 +1,257 @@
+
+        # Attempt to evaluate value as a literal (e.g. '1' => 1, ) If the value
+        # is just a string, is None, or is empty, it will raise an error. These
+        # should be considered varchars.
+        # E.g.
+        # "" => SyntaxError
+        # "anystring" => ValueError
+        try:
+            val_lit = ast.literal_eval(str(value))
+        except (SyntaxError, ValueError):
+            return self.VARCHAR
+
+        # Exit early if it's None
+        # is_valid_sql_num(None) == False
+        # instead of defaulting to varchar (which is the next test)
+        # return the compare type
+        if val_lit is None:
+            return cmp_type
+
+        # Make sure that it is a valid integer
+        # Python accepts 100_000 as a valid form of 100000,
+        # however a sql engine may throw an error
+        if not self.is_valid_sql_num(value):
+            return self.VARCHAR
+
+        type_lit = type(val_lit)
+
+        # If a float, stop here
+        # float is highest after varchar
+        if type_lit == float or cmp_type == self.FLOAT:
+            return self.FLOAT
+
+        # The value is very likely an int
+        # let's get its size
+        # If the compare types are empty and use the types of the current value
+        if type_lit == int and cmp_type in (self.INT_TYPES + [None, ""]):
+
+            # Use smallest possible int type above TINYINT
+            if (self.SMALLINT_MIN < val_lit < self.SMALLINT_MAX):
+                return self.get_bigger_int(self.SMALLINT, cmp_type)
+            elif (self.MEDIUMINT_MIN < val_lit < self.MEDIUMINT_MAX):
+                return self.get_bigger_int(self.MEDIUMINT, cmp_type)
+            elif (self.INT_MIN < val_lit < self.INT_MAX):
+                return self.get_bigger_int(self.INT, cmp_type)
+            else:
+                return self.BIGINT
+
+        # Need to determine who makes it all the way down here
+        return cmp_type
+
+    def format_column(self, col, index=None, replace_chars=None,
+                      col_prefix=None):
+        """Format the column to meet database contraints.
+
+        Formats the columns as follows:
+            1. Coverts to lowercase (if case insensitive)
+            2. Strips leading and trailing whitespace
+            3. Replaces invalid characters
+            4. Renames if in reserved words
+        `Args`:
+            col: str
+                The column to format.
+            index: int
+                (Optional) The index of the column. Used if the column is empty.
+            replace_chars: dict
+                A dictionary of invalid characters and their replacements. If
+                ``None`` uses {" ": "_"}
+            col_prefix: str
+                The prefix to use when the column is empty or starts with an
+                invalid character.
+        `Returns`:
+            str
+                The formatted column.
+        """
+        REPLACE_CHARS = {" ": "_"}
```

Comment:
Yeah that makes sense. I think I didn't throw it in there because I was hoping to allow the user to change it when calling this method. I think we can do both.

I'll make the change.